duration.go

```diff
@@ -117,12 +119,8 @@ func (d Duration) MarshalYAML() (interface{}, error) {
 }
 
 // UnmarshalYAML implements the yaml.Unmarshaler interface.
-func (d *Duration) UnmarshalYAML(unmarshal func(interface{}) error) error {
-	var s string
-	if err := unmarshal(&s); err != nil {
-		return err
-	}
-	dur, err := ParseDuration(s)
+func (d *Duration) UnmarshalYAML(node *yaml.Node) error {
+	dur, err := ParseDuration(node.Value)
```

Comment:
Damn, I was wrong! Sorry, but can you please revert this change and use `yaml.v2` everywhere?